adr/ADR-33.md

```diff
@@ -0,0 +1,236 @@
+# Service API
+
+| Metadata | Value                 |
+| -------- | --------------------- |
+| Date     | 2022-11-23            |
+| Author   | @aricart              |
+| Status   | Partially Implemented |
+| Tags     | client                |
+
+## Context and Problem Statement
+
+Simplify the development of NATS micro-services.
+
+The design goal of the API is to reduce development to having a complexity
+similar to that of writing a NATS subscription, but then by using simple
+configuration, allow the specified metadata to allow for standardization of
+discovery and observability.
+
+## Design
+
+Service configuration relies on the following:
```

Comment:
We need to specify which of those fields are mandatory and which are optional.

disregarding handlers for now, which provide functionality, not a config, for the config part, `name` and `version` should be mandatory.

Reply:
done